iceoryx2/src/port/publisher.rs

```diff
@@ -471,7 +471,6 @@ impl<'a, Service: service::Service, MessageType: Debug> UninitLoan<MessageType>
     for Publisher<'a, Service, MessageType>
 {
     fn loan_uninit(&self) -> Result<SampleMut<MaybeUninit<MessageType>>, PublisherLoanError> {
-        self.retrieve_returned_samples();
```

Comment:
This needs to be done in `loan_uninit` as well as `send_impl` else you have the exact opposite behavior and the loan might fail although the retrieve channels have free chunks.